clickhouse_driver/columns/datecolumn.py

```diff
@@ -11,7 +11,12 @@ class DateColumn(FormatColumn):
     epoch_start = date(1970, 1, 1)
 
     def before_write_item(self, value):
-        return (value - self.epoch_start).days
+        if not isinstance(value, date):
+            value = value.date()
+        diff = (value - self.epoch_start).days
+        if diff > (0x7fff * 2 + 1) or diff < 0:
```

Comment:
According to [docs](https://clickhouse.yandex/docs/en/single/#date) the final fully-supported year is 2105. The upper threshold defined by a constant at the compilation stage. Some servers has 2105 as max year some has different max value. But here he have

```python
In [5]: date(1970, 1, 1) + timedelta(0x7fff * 2 + 1)
Out[5]: datetime.date(2149, 6, 6)
```

It seems that is not correct to put this date-range validation here. Because it's precompiled.

Reply:
My bad, I remembered: why I put this validation. I got runtime error, described in #74: 
`clickhouse_driver.errors.TypeMismatchError: Code: 53. Type mismatch in VALUES section. Repeat query with types_check=True for detailed info. Column DEP_DATE_LOC: ushort format requires 0 <= number <= (0x7fff * 2 + 1)`
Also, You are right about hardcoding this value, but we need to check this value due error above... Maybe, we need to comment this stuff or?...

Reply:
It seems that 2015 year is [hardcoded right now](https://github.com/yandex/ClickHouse/blob/bb65c9dfd44e6b94c9eb64d447047441e2030601/libs/libcommon/include/common/DateLUTImpl.h#L14). You was right about range validation on the client. It should be within short int range. If ClickHouse server doesn't accept some date column value `0000-00-00` will be inserted.

All in all this validation code is OK. But, please put `0x7fff * 2 + 1` into class level like `epoch_start`.

Reply:
Okay, no problem.